YIF.Core.Service/Concrete/Services/DisciplineService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using YIF.Core.Domain.ServiceInterfaces;
+using YIF.Core.Data.Interfaces;
+using YIF.Core.Data.Entities;
+using YIF.Core.Domain.DtoModels.EntityDTO;
+using AutoMapper;
+using System.Resources;
+using YIF.Core.Domain.ApiModels.RequestApiModels;
+using System.Threading.Tasks;
+using YIF.Core.Domain.ApiModels.ResponseApiModels;
+using SendGrid.Helpers.Errors.Model;
+using YIF.Core.Domain.ApiModels.Validators;
+
+namespace YIF.Core.Service.Concrete.Services
+{
+    public class DisciplineService: IDisciplineService
+    {
+        private IDisciplineRepository<Discipline, DisciplineDTO> _disciplineRepository;
+        private readonly ResourceManager _resourceManager;
+        private readonly IMapper _mapper;
+        
+
+        public DisciplineService(
+            IDisciplineRepository<Discipline, DisciplineDTO> disciplineRepository,
+            ResourceManager resourceManager,
+            IMapper mapper
+        )
+        {
+            _disciplineRepository = disciplineRepository;
+            _resourceManager = resourceManager;
+            _mapper = mapper;
+        }
+
+        public async Task<ResponseApiModel<DescriptionResponseApiModel>> AddDiscipline(DisciplinePostApiModel disciplinePostApiModel)
+        {
+            var result = new ResponseApiModel<DescriptionResponseApiModel>();
+            
+            var discipline = _mapper.Map<Discipline>(disciplinePostApiModel);
```

Comment:
you should map it to DTO model

Reply:
I can't because my method Add in DisciplineRepository accepts Discipline, not DisciplineDTO as all Add methods accepts entity not DTO  in other repositories.

Reply:
yeah, but you can map it there too.